dags/data_dumps.py

```diff
@@ -0,0 +1,62 @@
+import datetime
+import airflow
+from airflow.models import DAG
+from operators.postgres_to_s3_transfer import PostgresToS3Transfer
+
+args = {
+    'owner': 'airflow',
+    'depends_on_past': False,
+    'start_date': datetime.datetime(2017, 3, 15),
+    'retries': 1,
+    'retry_delay': datetime.timedelta(minutes=10),
+}
+
+dag = DAG(
+    dag_id='data_dumps',
+    default_args=args,
+    max_active_runs=1,
+    schedule_interval='@monthly'
+)
+
+tables_to_dump = [
+    'conditions',
+    'document_categories',
+    'documents',
+    'fda_applications',
+    'fda_approvals',
+    'files',
+    'interventions',
+    'knex_migrations',
+    'knex_migrations_id_seq',
+    'knex_migrations_lock',
+    'locations',
+    'organisations',
+    'persons',
+    'publications',
+    'records',
+    'risk_of_bias_criterias',
+    'risk_of_biases',
+    'risk_of_biases_risk_of_bias_criterias',
+    'sources',
+    'trials',
+    'trials_conditions',
+    'trials_documents',
+    'trials_interventions',
+    'trials_locations',
+    'trials_organisations',
+    'trials_persons',
+    'trials_publications',
+]
+datastore_http = airflow.hooks.BaseHook.get_connection('datastore_http')
+DUMP_API_URL = '{base}{endpoint}'.format(
+    base=datastore_http.host.replace('http://', 's3://'),
+    endpoint='/dumps/opentrials-api-{{ end_date }}.dump'
```

Comment:
Airflow has a concept of templated variables (using Jinja). Here's a list of the available variables: https://pythonhosted.org/airflow/code.html#default-variables

The DAG/Task are immutable and should be completely defined on the code, but sometimes we want to change some variables depending on the run. For example, here I want the URL to be composed of the actual date the code was run. The variable name `end_date` is quite confusing here. In Airflow, `end_date == ds == execution_date (formatted as YYYY-MM-DD)`.